### PR TITLE
Update specification to include development profiles

### PR DESCRIPTION
**Pull request by dmathieu**
This updates the OTLP specification to have appropriate mentions of the profiles signal (and mention it's experimental).

### CODE REVIEW

**Review by christos68k — APPROVED** [review]

File: docs/specification.md [hunk 1]
```diff
@@ -92,7 +95,8 @@ telemetry data using unary requests using
 [Export*ServiceRequest](https://github.com/open-telemetry/opentelemetry-proto)
 messages ([ExportLogsServiceRequest](../opentelemetry/proto/collector/logs/v1/logs_service.proto) for logs,
 [ExportMetricsServiceRequest](../opentelemetry/proto/collector/metrics/v1/metrics_service.proto) for metrics,
-[ExportTraceServiceRequest](../opentelemetry/proto/collector/trace/v1/trace_service.proto) for traces).
+[ExportTraceServiceRequest](../opentelemetry/proto/collector/trace/v1/trace_service.proto) for traces),
```
Comment (tigrannajaryan) [inline docs/specification.md:98]: I think there is an extra closing round bracket.
```suggestion
[ExportTraceServiceRequest](../opentelemetry/proto/collector/trace/v1/trace_service.proto) for traces,
```

**Review by tigrannajaryan — APPROVED** [review]
LGTM, just need to fix one round bracket.

**Review by mx-psi — APPROVED** [review]

File: docs/specification.md [hunk 2]
```diff
@@ -456,6 +461,10 @@ the request body is a Protobuf-encoded `ExportMetricsServiceRequest` message.
 The default URL path for requests that carry log data is `/v1/logs` and the
 request body is a Protobuf-encoded `ExportLogsServiceRequest` message.
 
+The default URL path for requests that carry profiling data is
+`/v1experimental/profiles` and the request body is a Protobuf-encoded
```
[the remark this answers is not in the document]
Comment (felixge) [reply docs/specification.md:465]: Given that the experimental status has been renamed to development in https://github.com/open-telemetry/opentelemetry-specification/pull/4061, should we also update any usage of the word "experimental" in the context of profiling with "development"?

**Review by felixge — APPROVED** [review]
LGTM, Thanks.

**Review by Kielek — APPROVED** [review]

File: docs/specification.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmathieu) [reply docs/specification.md:465]: I'm not opposed to doing that. It's easy to do here, but would we also want to move the protobuf definitions to a `v1development` path?

[the remark this answers is not in the document]
Comment (felixge) [reply docs/specification.md:465]: Seems like the consistent thing to do. cc @open-telemetry/profiling-maintainers

[the remark this answers is not in the document]
Comment (dmathieu) [reply docs/specification.md]: See https://github.com/open-telemetry/opentelemetry-proto/pull/585

**Comment by dmathieu** [comment]
We're waiting for this change to move forward with the collector OTLP receiver. Could we get more reviews/a merge?

**Pull request merged**